### ISSUE TITLE
Keyboard navigation (up/down arrows) don't always work correctly when navigating repository list after adding filter

### ISSUE THREAD

**Issue by jsunsawyer** [desktop/desktop#11750]
### Describe the bug

After typing `CMD + T`, then typing a filter, you can sometimes not use the down arrow to select your desired repository.

This is inconsistent, but I've found a few ways to reproduce this unexpected behavior.

1. Simply press `CMD + T` (Don't type a filter)
2. Press `down arrow` once
3. Press `up arrow` (focus should be in the Filter)
4. Press `down arrow` (focus is stuck in the Filter)

After a bit more testing, it seems that the reverse can also be the case, where the cursor can get stuck in Filter and the up arrow can't move to the bottom of the repository list.

Originally, I though this issue stemmed from certain filters, where my narrowed list of repositories included ones that had been cloned to multiple folders, using the same name.

Sometimes there's an issue where, when navigating filtered repos of the same name, the focus gets stuck as if it were redirecting to the first one, preventing me from navigating further down my list with arrow keys.

### Version & OS

Github Desktop 2.6.5
macOS 10.15.76

**Comment by tidy-dev** [comment]
Thanks @jsunsawyer for the report! I was able to reproduce this as well and am marking as a bug.

**Comment by jsunsawyer** [comment]
Still experiencing this issue.

**Comment by timrspratt** [comment]
I too have been experiencing this issue for a while now, very frustrating when trying to navigate the repository list 😞

**Comment by timrspratt** [comment]
Hey, this still is an issue and seems to occur following a search of the repository list

**Comment by dtdesign** [comment]
Still happens to this day on macOS 13.5. I’m working with a large set of different repositories and this issue bites me once every one or two days.

I haven’t been able to reproduce this issue on purpose, but I’m somewhat certain that it appears to happen when I’m switching the repository using a filter that yields more results. This might indicate that the internal count of the matches is outdated/out of sync and thus causes the selection to wrap because it believes that it has reached the end of the list.

**Comment by sboerrigter** [comment]
I think https://github.com/desktop/desktop/issues/14447 is a related (or duplicate) issue with a bit more context. Maybe that helps to narrow it down.
